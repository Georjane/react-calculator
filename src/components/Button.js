/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Button.css';

export default function Button({
  name, clickHandler, color, wide,
}) {
  return <button className={color ? 'orange' : 'buttons'} type="button" onClick={clickHandler} style={{ width: wide ? '350px' : '175px' }}>{ name }</button>;
}

Button.propTypes = {
  name: PropTypes.string,
  clickHandler: PropTypes.func,
};

Button.defaultProps = {
  name: '',
  clickHandler: () => '',
};
