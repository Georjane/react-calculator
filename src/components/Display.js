import PropTypes from 'prop-types';

const Display = function (props) {
  return (
    <div className="Display">
      Display Component has result property:
      $
      {props.result}
    </div>
  );
};

Display.PropTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
