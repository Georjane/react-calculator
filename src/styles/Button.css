.buttons,
.orange {
  /* width: 175px; */
  height: inherit;
  color: black;
  border: 1px rgb(182, 179, 179) solid;
  font-size: 24px;
}

.orange {
  background-color: #ec903c;
}
