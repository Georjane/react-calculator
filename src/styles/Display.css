.display {
  background-color: #858694;
  height: 100px;
  color: white;
  font-weight: bold;
  padding-right: 20px;
  font-size: 24px;
  text-align: right;
}
