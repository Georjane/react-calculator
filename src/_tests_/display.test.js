import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';
import Button from '../components/Button';

it('renders button correctly', () => {
  const tree = renderer.create(<Button name="AC" clickHandler={jest.fn()} />);
  expect(tree).toMatchSnapshot();
});

it('renders display correctly', () => {
  const tree = renderer.create(<Display result="0" next="" />);
  expect(tree).toMatchSnapshot();
});
