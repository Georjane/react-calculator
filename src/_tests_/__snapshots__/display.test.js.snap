// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<button
  className="buttons"
  onClick={[MockFunction]}
  style={
    Object {
      "width": "175px",
    }
  }
  type="button"
>
  AC
</button>
`;

exports[`renders display correctly 1`] = `
<div
  className="display"
>
  <h2>
    0
  </h2>
</div>
`;
